frame/contracts/src/migration.rs

```diff
@@ -0,0 +1,275 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2018-2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{BalanceOf, CodeHash, Config, Pallet, TrieId, Weight};
+use codec::{Decode, Encode};
+use frame_support::{
+	codec, generate_storage_alias,
+	pallet_prelude::*,
+	storage::migration,
+	traits::{Get, PalletInfoAccess},
+	Identity, Twox64Concat,
+};
+use sp_std::{marker::PhantomData, prelude::*};
+
+/// Wrapper for all migrations of this pallet, based on `StorageVersion`.
+pub fn migrate<T: Config>() -> Weight {
+	use frame_support::traits::StorageVersion;
+
+	let version = StorageVersion::get::<Pallet<T>>();
+	let mut weight: Weight = 0;
+
+	if version < 4 {
+		weight = weight.saturating_add(v4::migrate::<T>());
+		StorageVersion::new(4).put::<Pallet<T>>();
+	}
+
+	if version < 5 {
+		weight = weight.saturating_add(v5::migrate::<T>());
+		StorageVersion::new(5).put::<Pallet<T>>();
+	}
+
+	if version < 6 {
+		weight = weight.saturating_add(v6::migrate::<T>());
+		StorageVersion::new(6).put::<Pallet<T>>();
+	}
+
+	if version < 7 {
+		weight = weight.saturating_add(v7::migrate::<T>());
+		StorageVersion::new(7).put::<Pallet<T>>();
+	}
+
+	weight
+}
```

Comment:
I don't think we actually need this logic?

Reply:
This is legacy code that will be wiped once we can correctly execute cosmwasm contracts. I didn't drop all ink related code yet